dataland-community-manager/src/test/kotlin/org/dataland/datalandcommunitymanager/services/SingleDataRequestEmailSenderTest.kt

```diff
@@ -0,0 +1,261 @@
+
+    @BeforeEach
+    fun setupSingleDataRequestEmailSender() {
+        val mockCompanyGetter: CompanyGetter = mock(CompanyGetter::class.java)
+        val singleDataRequestEmailBuilder = SingleDataRequestEmailBuilder(
+            proxyPrimaryUrl,
+            senderEmail,
+            senderName,
+            mockCompanyGetter,
+        )
+        val singleDataRequestInternalEmailBuilder = SingleDataRequestInternalEmailBuilder(
+            senderEmail,
+            senderName,
+            semicolonSeparatedInternalReceiverEmails,
+            semicolonSeparatedInternalCcEmails,
+            mockCompanyGetter,
+        )
+        mockEmailSender = mock(EmailSender::class.java)
+        singleDataRequestEmailSender = SingleDataRequestEmailSender(
+            mockEmailSender,
+            singleDataRequestEmailBuilder,
+            singleDataRequestInternalEmailBuilder,
+        )
+        val mockCompanyInformation = mock(CompanyInformation::class.java)
+        `when`(mockCompanyInformation.companyName).thenReturn(companyName)
+        `when`(mockCompanyGetter.getCompanyInfo(companyIdentifier)).thenReturn(mockCompanyInformation)
+    }
+
+    @Test
+    fun `validate that no email is sent if there are no reporting periods provided`() {
+        singleDataRequestEmailSender.sendSingleDataRequestEmails(
+            mockRequesterAuthentication,
+            SingleDataRequest(
+                "unused",
+                dataType,
+                listOfReportingPeriods = listOf(),
+                contactList = listOf("receiver@abc.de", "otherreceiver@something.else"),
+                message = defaultMessage,
+            ),
+            DataRequestCompanyIdentifierType.Isin,
+            companyIdentifier,
+        )
+        assertNumEmailsSentEquals(0)
+    }
+
+    @Test
+    fun `validate that an internal email is sent if there is no Dataland company ID provided`() {
+        expectSentEmailsToMatchInternalEmail(
+            DataRequestCompanyIdentifierType.Isin,
+        )
+        singleDataRequestEmailSender.sendSingleDataRequestEmails(
+            mockRequesterAuthentication,
+            SingleDataRequest(
+                "unused",
+                dataType,
+                listOfReportingPeriods = reportingPeriods,
+                contactList = listOf("receiver@abc.de", "otherreceiver@something.else"),
+                message = defaultMessage,
+            ),
+            DataRequestCompanyIdentifierType.Isin,
+            companyIdentifier,
+        )
+        assertNumEmailsSentEquals(1)
+    }
+
+    @Test
+    fun `validate that an internal email is sent if there are no contact provided`() {
+        expectSentEmailsToMatchInternalEmail(
+            DataRequestCompanyIdentifierType.DatalandCompanyId,
+            true,
+        )
+        singleDataRequestEmailSender.sendSingleDataRequestEmails(
+            mockRequesterAuthentication,
+            SingleDataRequest(
+                "unused",
```

Comment:
wouldn't be the test case "more" valid if a valid companyId was provided, so the only missing information is the empty contact list?
i mean, in theory the contactList feature could be broken an the email still would not be sent

Reply:
There is no check for the company format inside this function. But I see your point.